app/models/bookmark.rb

```diff
@@ -46,7 +46,14 @@ def self.auto_delete_preferences
   validates :name, length: { maximum: 100 }
 
   def unique_per_post_for_user
-    if Bookmark.exists?(user_id: user_id, post_id: post_id)
+    is_first_post = Post.where(id: post_id).pluck_first(:post_number) == 1
+    exists = if is_first_post
+      Bookmark.exists?(user_id: user_id, post_id: post_id, for_topic: for_topic)
+    else
+      Bookmark.exists?(user_id: user_id, post_id: post_id)
+    end
```

Comment:
I am thinking if a new column is required here. What would happen if in the case of a global topic bookmark we would skip setting post_id? So global bookmark is where(post_id: nil). 

It might be a bit less confusing in the future. Like why did I have a post_id set when I bookmarked the topic?

Reply:
We can describe behaviour with methods like 
```
def for_topic?
  post_id.blank?
 end
 ```
 and scopes etc.